azurelinuxagent/pa/deprovision/ubuntu.py

```diff
@@ -40,3 +40,13 @@ def del_resolv(self, warnings, actions):
             files_to_del = ["/etc/resolvconf/resolv.conf.d/tail",
                             "/etc/resolvconf/resolv.conf.d/original"]
             actions.append(DeprovisionAction(fileutil.rm_files, files_to_del))
+
+
+class Ubuntu1804DeprovisionHandler(UbuntuDeprovisionHandler):
+    def __init__(self):
+        super(Ubuntu1804DeprovisionHandler, self).__init__()
+
+    def del_resolv(self, warnings, actions):
```

Comment:
Is there any reason to keep the prior behavior?

Reply:
We still want the other actions implemented by the base handler.

Reply:
Sorry for being unclear.  We are warning about something we used to do that was bad.  If this is bad we shouldn't be doing it now or in previous versions.  I believe this is the intent of #855.  It seems odd to warn about something you previously did illegally especially considering it *breaks* the VM.

You could also argue that we will address #855 in a later version as a single PR.

Reply:
Changing established behavior in 1604 is highly risky, and I do not believe we should take the chance of breaking existing patterns. 1804 is a good opportunity to make that change though, and issuing a warning seems sensible if people have taken a dependency on that broken behavior, so they can change it. If you are arguing this should be an info message instead, sure.

#855 is specifically about cloud-init scripts, which is a sub-class of this problem. Aside from these two instances, I am not aware of any issues or asks in this area.